tests/e2e/e2e_test.go

```diff
@@ -160,6 +160,67 @@ func (s *IntegrationTestSuite) TestAddToExistingLock() {
 	chainA.LockAndAddToExistingLock(sdk.NewInt(1000000000000000000), fmt.Sprintf("gamm/pool/%d", poolId), lockupWalletAddr, lockupWalletSuperfluidAddr)
 }
 
+// Test03TWAP tests TWAP by creating a pool, performing a swap.
+// These two operations should create TWAP records.
+// Then, we wait until the epoch for the records to be pruned.
+// The records are guranteed to be pruned at the next epoch
+// because twap keep time = epoch time / 4 and we use a timer
+// to wait for at least the twap keep time.
+// TODO: implement querying for TWAP, once such queries are exposed:
+// https://github.com/osmosis-labs/osmosis/issues/2602
+func (s *IntegrationTestSuite) Test03TWAP() {
+	const (
+		poolFile   = "nativeDenomPool.json"
+		walletName = "swap-exact-amount-in-wallet"
+
+		coinIn       = "101stake"
+		minAmountOut = "99"
+		denomOut     = "uosmo"
+
+		epochIdentifier = "day"
+	)
+
+	chainA := s.configurer.GetChainConfig(0)
+	chainANode, err := chainA.GetDefaultNode()
+	s.NoError(err)
+
+	// Triggers the creation of TWAP records.
+	poolId := chainANode.CreatePool(poolFile, initialization.ValidatorWalletName)
+	swapWalletAddr := chainANode.CreateWallet(walletName)
+
+	chainANode.BankSend(coinIn, chainA.NodeConfigs[0].PublicAddress, swapWalletAddr)
+	heightBeforeSwap := chainANode.QueryCurrentHeight()
+
+	// Triggers the creation of TWAP records.
+	chainANode.SwapExactAmountIn(coinIn, minAmountOut, fmt.Sprintf("%d", poolId), denomOut, swapWalletAddr)
+
+	keepPeriodCountDown := time.NewTimer(initialization.TWAPPruningKeepPeriod)
+
+	// Make sure still producing blocks.
+	chainA.WaitUntilHeight(heightBeforeSwap + 3)
+
+	if !s.skipUpgrade {
+		// TODO: we should reduce the pruning time in the v11
+		// genesis to make this test run faster
+		// Currenty, we are testing the upgrade from v11 to v12,
+		// the pruning time is set to whatever is in the upgrade
+		// handler (two days). Therefore, we cannot reasonably
+		// test twap pruning post-upgrade.
+		s.T().Skip("skipping TWAP Pruning test. This can be re-enabled post v12")
+	}
```

Comment:
Should we just keep this at the top of the test for now, since we have shown that this will work up to this point? Only because this will add a little bit of time to CI without actually testing twap

Reply:
It tests twap by creating a pool and a swap. This is the core of the test - making sure that post-swap TWAP logic does not break